packages/next/build/check-next-script-import.ts

```diff
@@ -0,0 +1,16 @@
+import { promises } from 'fs'
+import { join } from 'path'
+
+export async function checkNextScriptImport(
+  pagePaths: string[],
+  pagesDir: string
+): Promise<boolean> {
+  for (const page of pagePaths) {
+    const fileContent = await promises.readFile(join(pagesDir, page), 'utf8')
+    if (/import[^}]*.*('|")next\/script('|")/.test(fileContent)) {
```

Comment:
Using `await` inside `for...of` loop degrades the performance by *a lot*. `Promise.all` should be used here:

```js
  await Promise.all(pagePaths.map(
    async (page) => {
      const fileContent = await promises.readFile(join(pagesDir, page), 'utf8')
    }
  ))
```

Reply:
Good call. Updated the PR to use `Promise.all` but I'm going to switch to using `swc` :)